pkg/i2gw/notifications/notifications.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifications
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	NotificationAggr = NotificationAggregator{Notifications: map[string][]Notification{}}
+}
+
+const (
+	InfoNotification    MessageType = "INFO"
+	WarningNotification MessageType = "WARNING"
+	ErrorNotification   MessageType = "ERROR"
+)
+
+type MessageType string
+
+type Notification struct {
+	Type           MessageType
+	Message        string
+	CallingObjects []client.Object
+}
+
+type NotificationAggregator struct {
+	mutex         sync.Mutex
+	Notifications map[string][]Notification
+}
+
+var NotificationAggr NotificationAggregator
+
+// DispatchNotification is used to send a notification to the NotificationAggregator
+func (na *NotificationAggregator) DispatchNotification(notification Notification, ProviderName string) {
+	na.mutex.Lock()
+	na.Notifications[ProviderName] = append(na.Notifications[ProviderName], notification)
+	na.mutex.Unlock()
+}
+
+// CreateNotificationTables takes all generated notifications and displays it in a tabular format based on provider
```

Comment:
if you decide to apply my comment about returning a a string from here (i.e return `t.Render()`) then dont forget to update this comment to "returns it in a tabular format"